<launch>
  <!-- debug print -->
  <param name="/gacm/debug" type="bool" value="true"/>
  <!-- true: display in the camera frame; false: in the lidar frame -->
  <param name="/gacm/display_frame_cam" type="bool" value="false"/>

  <arg name="config_path" default = "$(find gacm)/config/graco_ugv.yaml" />
  <arg name="config_path1" default = "$(find gacm)/config/graco_uav_orb.yaml" />
  <arg name="config_path2" default = "$(find gacm)/config/graco_uav.yaml" />
  <arg name="config_path3" default = "$(find gacm)/config/graco_ugv.yaml" />
  <arg name="config_path4" default = "$(find gacm)/config/graco_ugv.yaml" />
  <arg name="gacm_path" default = "$(find gacm)/" />

  <node pkg="gacm" type="gacm_uninode" name="gacm_uninode" output="screen">
    <param name="config_file" type="string" value="$(arg config_path)"/>
    <param name="config_file1" type="string" value="$(arg config_path1)"/>
    <param name="config_file2" type="string" value="$(arg config_path2)"/>
    <param name="config_file3" type="string" value="$(arg config_path3)"/>
    <param name="config_file4" type="string" value="$(arg config_path4)"/>
    <param name="gacm_folder" type="string" value="$(arg gacm_path)" />

    <rosparam param="left_image_directories">["/workspaces/dataset/graco/euroc_fmt/ground-06/mav0/cam0/data","/workspaces/dataset/graco/euroc_fmt/aerial-08-25m/mav0/cam0/data"]</rosparam>
    <rosparam param="point_cloud_directories">["/workspaces/dataset/graco/euroc_fmt/ground-06/mav0/point0/data","/workspaces/artifacts/ORB_SLAM3/graco/aerial-08-25m/points"]</rosparam>
    <rosparam param="odometry_files">["","/workspaces/artifacts/ORB_SLAM3/graco/aerial-08-25m/CameraTrajectory.txt"]</rosparam>
  </node>

  <arg name="rviz" default="true" />
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find gacm)/rviz_cfg/gacm.rviz" />
  </group>
</launch>
